ERROR - 2023-12-17 21:24:32 --> Error connecting to the database: mysqli_sql_exception: No connection could be made because the target machine actively refused it in C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php:184
Stack trace:
#0 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php(184): mysqli->real_connect('127.0.0.1', 'root', Object(SensitiveParameterValue), 'db_tracerstudy', 3306, '', 0)
#1 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\BaseConnection.php(392): CodeIgniter\Database\MySQLi\Connection->connect(false)
#2 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\BaseConnection.php(575): CodeIgniter\Database\BaseConnection->initialize()
#3 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1615): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#4 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Model.php(871): CodeIgniter\Database\BaseBuilder->get()
#5 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\app\Models\DataTables.php(59): CodeIgniter\Model->__call('get', Array)
#6 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\app\Controllers\AlumniAdmin.php(141): App\Models\DataTables->get_datatables('alumni', Array, Array, Array, Array)
#7 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(942): App\Controllers\AlumniAdmin->ajax()
#8 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\AlumniAdmin))
#9 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#10 C:\Users\wanad\xampp8\htdocs\traceralumnisi\admin\index.php(79): CodeIgniter\CodeIgniter->run()
#11 {main}

Next CodeIgniter\Database\Exceptions\DatabaseException: No connection could be made because the target machine actively refused it in C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\MySQLi\Connection.php:229
Stack trace:
#0 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\BaseConnection.php(392): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\BaseConnection.php(575): CodeIgniter\Database\BaseConnection->initialize()
#2 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Database\BaseBuilder.php(1615): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#3 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\Model.php(871): CodeIgniter\Database\BaseBuilder->get()
#4 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\app\Models\DataTables.php(59): CodeIgniter\Model->__call('get', Array)
#5 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\app\Controllers\AlumniAdmin.php(141): App\Models\DataTables->get_datatables('alumni', Array, Array, Array, Array)
#6 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(942): App\Controllers\AlumniAdmin->ajax()
#7 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\AlumniAdmin))
#8 C:\Users\wanad\xampp8\htdocs\traceralumnisi-ci4\admin-ci4\vendor\codeigniter4\framework\system\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 C:\Users\wanad\xampp8\htdocs\traceralumnisi\admin\index.php(79): CodeIgniter\CodeIgniter->run()
#10 {main}
CRITICAL - 2023-12-17 21:24:32 --> Unable to connect to the database.
Main connection [MySQLi]: No connection could be made because the target machine actively refused it
in SYSTEMPATH\Database\BaseConnection.php on line 428.
 1 SYSTEMPATH\Database\BaseConnection.php(575): CodeIgniter\Database\BaseConnection->initialize()
 2 SYSTEMPATH\Database\BaseBuilder.php(1615): CodeIgniter\Database\BaseConnection->query('SELECT *
FROM `alumni`
WHERE `id_alumni` != :id_alumni:
ORDER BY `nim_alumni` DESC
 LIMIT 20', [...], false)
 3 SYSTEMPATH\Model.php(871): CodeIgniter\Database\BaseBuilder->get()
 4 APPPATH\Models\DataTables.php(59): CodeIgniter\Model->__call('get', [])
 5 APPPATH\Controllers\AlumniAdmin.php(141): App\Models\DataTables->get_datatables('alumni', [...], [...], [...], [...])
 6 SYSTEMPATH\CodeIgniter.php(942): App\Controllers\AlumniAdmin->ajax()
 7 SYSTEMPATH\CodeIgniter.php(502): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\AlumniAdmin))
 8 SYSTEMPATH\CodeIgniter.php(361): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(79): CodeIgniter\CodeIgniter->run()
